common/remote/sockfile.cpp

```diff
@@ -4280,6 +4331,77 @@ class CRemoteFileServer : public CInterface, implements IRemoteFileServer, imple
         return false;
     }
 
+    bool cmdSetThrottle(MemoryBuffer & msg, MemoryBuffer & reply,CRemoteClientHandler &client)
+    {
+        CriticalBlock b(setThrottleCrit);
+        try
+        {
+            unsigned _throttleLimit, _throttleDelayMs, _throttleCPULimit;
+            msg.read(_throttleLimit);
+            msg.read(_throttleDelayMs);
+            msg.read(_throttleCPULimit);
+            int delta = 0;
+            if (_throttleLimit)
+            {
+                if (disabledThrottleLimit) // if transitioning from disabled to some throttling
+                {
+                    assertex(0 == throttleLimit);
+                    delta = _throttleLimit - disabledThrottleLimit; // + or -
+                    disabledThrottleLimit = 0;
+                }
+                else
+                    delta = _throttleLimit - throttleLimit; // + or -
+            }
+            else
```

Comment:
else if (disabledThrottleLimit != 0)

otherwise setting to 0 2 times in a row will lose the value.